bigchaindb/tendermint/lib.py

```diff
@@ -90,6 +97,57 @@ def store_bulk_transactions(self, transactions):
             backend.query.store_assets(self.connection, assets)
         return backend.query.store_transactions(self.connection, txns)
 
+    def update_utxoset(self, transaction):
+        """ .. todo: docs """
+        spent_outputs = [
+            spent_output for spent_output in transaction.spent_outputs
+        ]
+        if spent_outputs:
+            self.delete_unspent_outputs(*spent_outputs)
+        self.store_unspent_outputs(
+            *[utxo._asdict() for utxo in transaction.unspent_outputs]
+        )
+
+    def get_utxoset_merkle_root(self):
+        """Returns the merkle root of the utxoset. This implies that
+        the utxoset is first put into a merkle tree.
+
+        For now, the merkle tree and its root will be computed each
+        time. This obviously is not efficient and a better approach
+        that limits the repetition of the same computation when
+        unnecesary should be sought. For instance, future optimizations
+        could simply re-compute the branches of the tree that were
+        affected by a change.
+
+        The transaction hash (id) and output index should be sufficient
+        to uniquely identify a utxo, and consequently only that
+        information from a utxo record is needed to compute the merkle
+        root. Hence, each node of the merkle tree should contain the
+        tuple (txid, output_index).
+
+        .. important: The leaves of the tree will need to be sorted in
+        some kind of lexicographical order.
```

Comment:
A partial lexographic order can be imposed by the using the block height and the transaction inside a block can be sequentially ordered?

Reply:
@vrde In my understanding if we have a deterministic lexographic order then we may be able to avoid calculating some parent hashes depening on where the update has been made. But if we  take a sorted hash list of leaves each time then we would have a different arrangements of leaves each time so the whole tree would need to be re-calculated.

Reply:
@kansi what you are saying is to use the block height as the first attribute to "sort" the unspents, correct? If so, I think it may improve performance, since it's easier to cache some "subtrees" of hashes. For now, I think @sbellem approach is OK, we can improve it later.

Reply:
@vrde yup, that is what I am suggestion. But thats an optimization we can implement later.